ledger/src/blockstore.rs

```diff
@@ -3281,21 +3262,45 @@ impl Blockstore {
                 })
                 .collect();
         let data_shreds = data_shreds?;
-        let last_shred = data_shreds.last().unwrap();
-        assert!(last_shred.data_complete() || last_shred.last_in_slot());
 
-        let deshred_payload = Shredder::deshred(&data_shreds).map_err(|e| {
-            BlockstoreError::InvalidShredData(Box::new(bincode::ErrorKind::Custom(format!(
-                "Could not reconstruct data block from constituent shreds, error: {e:?}"
-            ))))
-        })?;
+        let mut entries = vec![];
+        for (start_index, end_index) in completed_ranges.iter() {
+            // The indices from completed_ranges refer to shred indices in the
+            // block; map those indices to indices within data_shreds
+            let start_index = (*start_index - total_start_index) as usize;
+            let end_index = (*end_index - total_start_index) as usize;
+            let range_shreds = &data_shreds[start_index..=end_index];
+
+            let last_shred = range_shreds.last().unwrap();
+            assert!(last_shred.data_complete() || last_shred.last_in_slot());
+
+            let deshred_payload = Shredder::deshred(range_shreds).map_err(|e| {
+                BlockstoreError::InvalidShredData(Box::new(bincode::ErrorKind::Custom(format!(
+                    "could not reconstruct entries buffer from shreds: {e:?}"
+                ))))
+            })?;
+
+            debug!("{:?} shreds in last FEC set", data_shreds.len());
```

Comment:
I realize this is and existing log just moved around, but maybe we can clarify this is the number of data* shreds?

Reply:
Done - gonna demote it to trace as well as that one could be quite noisy